packages/system/src/util.js

```diff
@@ -0,0 +1,113 @@
+'use strict'
+
+import invariant from 'invariant'
+
+export const partial = (f, ...args) => f.bind(null, ...args)
+export const complement = f => (...args) => !f(...args)
+
+export const prop = (p, obj) => obj[p]
+export const propd = (p, dflt, obj) => (obj[p] != null ? obj[p] : dflt)
+export const path = (path, obj) => {
+  let c = obj
+
+  for (const prop of path) {
+    c = c[prop]
+    if (c == null) return undefined
+  }
+
+  return c
+}
+
+export const map = f => coll => coll.map(f)
+export const assoc = (p, value, obj) => ({ ...obj, [p]: value })
+export const update = (p, fn, obj) => assoc(p, fn(prop(p, obj)), obj)
+
+export const reject = (pred, coll) => coll.filter(complement(pred))
+export const flatten = coll =>
+  coll.length <= 1 ? coll : coll[0].concat(...coll.slice(1))
+
+export const isNil = value => value == null
+
+export const identity = x => x
+
+export const pipe = (...fs) => x => fs.reduce((r, f) => f(r), x)
+
+export const flow = (value, ...fns) => {
```

Comment:
Isn't this the same as [this](https://github.com/netceteragroup/skele/blob/7a6152ce7b15a5ea2d89ca61d3a534a29a1666d1/packages/classic/src/data/index.js#L12)? Maybe it's a good idea to have a `util` or `data` package, which other packages will depend on. It could contain shared functionalities such as this one.